src/java/com/zimbra/qa/selenium/projects/admin/ui/PageManageAccounts.java

```diff
@@ -139,52 +138,173 @@ public void zNavigateTo(String treeItem) {
 
-		int m = 50;
 		if (count >= 50) {
-			for (int a1 = 1; a1 <= 10; a1++) { 
-				String p0  = rowsLocator + ":nth-child("+m+")";				
-				if (this.sIsElementPresent(p0)) {					
-					sClickAt(p0,"");
+			
+			for (int accountPaging = 1; accountPaging <= 30; accountPaging++) {
+				
+				String pageCounter = rowsLocator + ":nth-child(" + scrollCounter + ")";
+				SleepUtil.sleepVerySmall();
+				if (this.sIsElementPresent(pageCounter)) {
+					sClickAt(pageCounter, "");
+					SleepUtil.sleepVerySmall();
 					this.zKeyboard.zTypeKeyEvent(KeyEvent.VK_DOWN);
-					m = m + 20;
+					scrollCounter = scrollCounter + 20;
 				} else {
 					break;
 				}
 			}
-
+			
+			accountLocator = "//div[@id='zl__ACCT_MANAGE']//div[starts-with(@id,'zli__DWT')]//td[starts-with(@id,'account_data_emailaddress_')]/nobr[.='" + accountEmailAddress + "']";
+			//accountLocator = "css=div[id^='zli__DWT'] td[id^='account_data_emailaddress']:contains('" + account + "')";
+			if (sIsElementPresent(accountLocator)) {
+				accountFound = true;
+			}
+			
+			if (accountFound.equals(true)) {
+				if (action == Action.A_LEFTCLICK) {
+					sClickAt(accountLocator, "");
+				} else if (action == Action.A_RIGHTCLICK) {
+					zRightClickAt(accountLocator, "");
+				}
+			} else {
+				throw new HarnessException("Couldn't find account " + accountEmailAddress + " after virtual paging.");
+			}
 		}
-		// xpath to the account
-		String accountLocator = "//div[@id='zl__ACCT_MANAGE']//div[starts-with(@id,'zli__DWT')]//td[starts-with(@id,'account_data_emailaddress_')]/nobr[.='" + account + "']";
 
-		if (action == Action.A_LEFTCLICK) {
-			sClick(accountLocator);
-		} else if(action == Action.A_RIGHTCLICK) {
-			zRightClick(accountLocator);
-		}
+		SleepUtil.sleepSmall();
 		return page;
 	}
 
 	@Override
-	public AbsPage zListItem(Action action, Button option, String item)
-			throws HarnessException {
-		return null;
+	public AbsPage zListItem(Action action, Button option, String accountEmailAddress) throws HarnessException {
+		
+		logger.info(myPageName() + " zListItem(" + action + ", " + option + ", " + accountEmailAddress + ")");
+
+		AbsPage page = null;
+		String accountLocator, optionLocator = null;
+		Boolean accountFound = false;
+		
+		SleepUtil.sleepMedium();
+		
+		if (action == Action.A_RIGHTCLICK) {
+			
+			((AppAdminConsole)MyApplication).zPageSearchResults.zAddSearchQuery(accountEmailAddress);
+			((AppAdminConsole)MyApplication).zPageSearchResults.zToolbarPressButton(Button.B_SEARCH);
+
+			// Get the list of displayed accounts
+			List<AccountItem> accounts = ((AppAdminConsole)MyApplication).zPageSearchResults.zListGetAccounts();
+
+			for (AccountItem a : accounts) {
+				logger.info("Looking for account " + accountEmailAddress + " found: " + a.getGEmailAddress());
+				if (accountEmailAddress.equals(a.getGEmailAddress())) {
+					accountFound = true;
+					break;
+				}
+			}
+			ZAssert.assertNotNull(accountFound, "Verify account is found in search results");

```

Comment:
ZAssert.assertTrue(accountFound, "Verify account is found in search result");

Reply:
Made this change too, thanks Jitesh.